### ISSUE TITLE
For object detection tasks, the FLIR dataset cannot achieve the accuracy published in the paper through training. 目标检测任务，FLIR数据集无法通过训练达到文章中公布的精度。

### ISSUE THREAD

**Issue by PatrickStar-Li** [PoTsui99/UniRGB-IR#12]
中文：
作者您好，我对您的工作十分感兴趣，不过现在遇到一些问题，还希望您能解答一下。
问题的详细描述：
FLIR数据集，经过训练精度仅可以达到以下的效果：
 Average Precision  (AP) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = 0.317
 Average Precision  (AP) @[ IoU=0.50      | area=   all | maxDets=1000 ] = 0.680
 Average Precision  (AP) @[ IoU=0.75      | area=   all | maxDets=1000 ] = 0.255
 Average Precision  (AP) @[ IoU=0.50:0.95 | area= small | maxDets=1000 ] = 0.155
 Average Precision  (AP) @[ IoU=0.50:0.95 | area=medium | maxDets=1000 ] = 0.400
 Average Precision  (AP) @[ IoU=0.50:0.95 | area= large | maxDets=1000 ] = 0.530
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = 0.441
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=300 ] = 0.441
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=1000 ] = 0.441
 Average Recall     (AR) @[ IoU=0.50:0.95 | area= small | maxDets=1000 ] = 0.317
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=medium | maxDets=1000 ] = 0.513
 Average Recall     (AR) @[ IoU=0.50:0.95 | area= large | maxDets=1000 ] = 0.625
04/12 08:13:49 - mmengine - INFO - bbox_mAP_copypaste: 0.317 0.680 0.255 0.155 0.400 0.530
04/12 08:13:49 - mmengine - INFO - Iter(val) [254/254]    coco/bbox_mAP: 0.3170  coco/bbox_mAP_50: 0.6800  coco/bbox_mAP_75: 0.2550  coco/bbox_mAP_s: 0.1550  coco/bbox_mAP_m: 0.4000  coco/bbox_mAP_l: 0.5300  data_time: 0.0809  time: 0.3872

而使用作者提供的已训练好的公开权重，可以达到文章中的精度，这说明数据集处理没有问题，评价指标也没有问题。
↓
 Average Precision  (AP) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = 0.451
 Average Precision  (AP) @[ IoU=0.50      | area=   all | maxDets=1000 ] = 0.827
 Average Precision  (AP) @[ IoU=0.75      | area=   all | maxDets=1000 ] = 0.427
 Average Precision  (AP) @[ IoU=0.50:0.95 | area= small | maxDets=1000 ] = 0.313
 Average Precision  (AP) @[ IoU=0.50:0.95 | area=medium | maxDets=1000 ] = 0.523
 Average Precision  (AP) @[ IoU=0.50:0.95 | area= large | maxDets=1000 ] = 0.640
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = 0.553
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=300 ] = 0.553
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=1000 ] = 0.553
 Average Recall     (AR) @[ IoU=0.50:0.95 | area= small | maxDets=1000 ] = 0.447
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=medium | maxDets=1000 ] = 0.613
 Average Recall     (AR) @[ IoU=0.50:0.95 | area= large | maxDets=1000 ] = 0.731
04/12 07:00:00 - mmengine - INFO - bbox_mAP_copypaste: 0.451 0.827 0.427 0.313 0.523 0.640

下面是我复现过程中遇到的问题，这是否导致了我复现不出来文章中的精度？
(1) 在train_od.sh中，设置了配置文件1024_v15_rgb-ir-feat-fusion-spm_crossAttn-GRU_4stage_8x1bs.py，考虑到我的计算资源没有这么多，我才用了size=1024，batchsize=2GPU*1的模式。另外，作者提供的train_od.sh文件中使用的配置文件是1024_v15_rgb-ir-feat-fusion-spm_crossAttn-GRU_staged_8x1bs.py，这跟4stage文件是否有区别？
(2) 在上述的配置文件1024_v15_rgb-ir-feat-fusion-spm_crossAttn-GRU_4stage_8x1bs.py中需要设置ViT的预训练模型，该文件中默认设置的是vitb_coco_IN1k_mae_coco_cascade-mask-rcnn_224x224_withClsToken_noRel.pth，而在huggingface项目中上传的vitb_IN1k_mae_coco_cascade-mask-rcnn_224x224_noClsToken_noRel.pth是作者提供的ViT的预训练模型。这个withClsToken和noClsToken是否有区别？
在这个情况下精度仍然跟一个Closed的issue一样。

十分感谢作者的答疑解惑。

EN:

Dear author, I am very interested in your work, but I have encountered some problems and hope you can help me.
Detailed description of the problem: With the FLIR dataset, the trained accuracy is only as follows:
 Average Precision  (AP) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = 0.317
 Average Precision  (AP) @[ IoU=0.50      | area=   all | maxDets=1000 ] = 0.680
 Average Precision  (AP) @[ IoU=0.75      | area=   all | maxDets=1000 ] = 0.255
 Average Precision  (AP) @[ IoU=0.50:0.95 | area= small | maxDets=1000 ] = 0.155
 Average Precision  (AP) @[ IoU=0.50:0.95 | area=medium | maxDets=1000 ] = 0.400
 Average Precision  (AP) @[ IoU=0.50:0.95 | area= large | maxDets=1000 ] = 0.530
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = 0.441
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=300 ] = 0.441
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=1000 ] = 0.441
 Average Recall     (AR) @[ IoU=0.50:0.95 | area= small | maxDets=1000 ] = 0.317
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=medium | maxDets=1000 ] = 0.513
 Average Recall     (AR) @[ IoU=0.50:0.95 | area= large | maxDets=1000 ] = 0.625
04/12 08:13:49 - mmengine - INFO - bbox_mAP_copypaste: 0.317 0.680 0.255 0.155 0.400 0.530
04/12 08:13:49 - mmengine - INFO - Iter(val) [254/254]    coco/bbox_mAP: 0.3170  coco/bbox_mAP_50: 0.6800  coco/bbox_mAP_75: 0.2550  coco/bbox_mAP_s: 0.1550  coco/bbox_mAP_m: 0.4000  coco/bbox_mAP_l: 0.5300  data_time: 0.0809  time: 0.3872

However, using the pre-trained public weights provided by the author, the accuracy reported in the paper can be achieved, indicating that the dataset processing and evaluation metrics are not the problem.
↓ 
Average Precision (AP) @[ IoU=0.50:0.95 | area= all | maxDets=100 ] = 0.451 
Average Precision (AP) @[ IoU=0.50 | area= all | maxDets=1000 ] = 0.827 
Average Precision (AP) @[ IoU=0.75 | area= all | maxDets=1000 ] = 0.427 
Average Precision (AP) @[ IoU=0.50:0.95 | area= small | maxDets=1000 ] = 0.313 
Average Precision (AP) @[ IoU=0.50:0.95 | area=medium | maxDets=1000 ] = 0.523 
Average Precision (AP) @[ IoU=0.50:0.95 | area= large | maxDets=1000 ] = 0.640 
Average Recall (AR) @[ IoU=0.50:0.95 | area= all | maxDets=100 ] = 0.553 
Average Recall (AR) @[ IoU=0.50:0.95 | area= all | maxDets=300 ] = 0.553 
Average Recall (AR) @[ IoU=0.50:0.95 | area= all | maxDets=1000 ] = 0.553 
Average Recall (AR) @[ IoU=0.50:0.95 | area= small | maxDets=1000 ] = 0.447 
Average Recall (AR) @[ IoU=0.50:0.95 | area=medium | maxDets=1000 ] = 0.613 
Average Recall (AR) @[ IoU=0.50:0.95 | area= large | maxDets=1000 ] = 0.731
04/12 07:00:00 - mmengine - INFO - bbox_mAP_copypaste: 0.451 0.827 0.427 0.313 0.523 0.640

Below are the problems I encountered during the reproduction process. Did this cause me to be unable to reproduce the accuracy in the article?
(1) In train_od.sh, I set the configuration file 1024_v15_rgb-ir-feat-fusion-spm_crossAttn-GRU_4stage_8x1bs.py. Considering that I do not have that many computing resources, I used the mode of size=1024, batchsize=2GPU*1. Additionally, the configuration file used in the `train_od.sh` file provided by the author is `1024_v15_rgb-ir-feat-fusion-spm_crossAttn-GRU_staged_8x1bs.py`. Is there any difference between this and the 4-stage file?
(2) In the configuration file `1024_v15_rgb-ir-feat-fusion-spm_crossAttn-GRU_4stage_8x1bs.py`, the ViT pre-trained model needs to be set. The default setting in this file is `vitb_coco_IN1k_mae_coco_cascade-mask-rcnn_224x224_withClsToken_noRel.pth`, while the `vitb_IN1k_mae_coco_cascade-mask-rcnn_224x224_noClsToken_noRel.pth` uploaded to the `huggingface` project is the ViT pre-trained model provided by the author. Is there any difference between `withClsToken` and `noClsToken`? In this case, the accuracy remains the same as a closed issue.

Many thanks to the author for answering my questions.

**Comment by PoTsui99** [comment]
* 针对 (1): 4stage 仅输入图像尺寸不同 (输入尺寸 768 像素, 用于消融实验). 
* 针对 (2): [CLS] token 在模型中并未使用, HF 上下载的权重对训练及推理表现无影响.

关于无法复现问题, 我的判断和建议: (1) batch 较小通常对指标无影响或有略微正向影响; (2) 可能要排查看权重是否正确加载, 可加断点在第一次 backprop 前看下模型权重和 pretrained 权重一致; (3) UniRGB-IR 是在 Pascal 和 Ampere 架构的 N 卡上训练, 不清楚显卡或 CUDA 版本不一致是否会导致精度问题, 可先采用云 GPU 等手段复现, 环境建议与 `environment.yml` 对齐.

感谢您对 UniRGB-IR 的关注.

**Comment by PatrickStar-Li** [comment]
很感谢作者的提醒，经过我的尝试，将图像尺寸调小后并设置了稍大的batchsize，同时设置了适配小batchsize的学习率，mAP@0.5:0.95的指标已经可以达到42.0，尚且还有差距，我再自己琢磨下。
